test-resources/headless-core-stub/lambdas/callback/src/services/callback-service.ts

```diff
@@ -0,0 +1,53 @@
+import { DynamoDBClient, QueryCommand } from "@aws-sdk/client-dynamodb";
+import { SessionItem } from "./session-item";
+
+export class CallBackService {
+    constructor(private readonly dynamoDbClient = new DynamoDBClient({ region: process.env.REGION })) {}
+
+    public async getSessionByAuthorizationCode(sessionTable: string, code: string): Promise<SessionItem> {
+        const sessionItemQuery = await this.dynamoDbClient.send(
```

Comment:
We'll need to add a re-try! It's querying a GSI

Reply:
Maybe not in this PR, will be future improvement but good to point out

Reply:
Any reason we can just wrap this in a for-loop e.g. 

```
let item;
for(i .. 3) {
  // existing logic
 if (count != 0) {
    item = get(0);
    break;
   }
}
if(!item) throw Error();
return item;
```

I'm thinking we do this now because we're updating tests after this and the QAs might end up adding re-tries over there.

Reply:
Similarly, other teams might be adopting this so we don't want known error scenarios being present if the fix the simple.

Reply:
Similarly, other teams might be adopting this so we don't want known error scenarios being present if the fix the simple.